src/test/java/uk/gov/pay/api/service/AgreementServiceTest.java

```diff
@@ -82,6 +83,25 @@ void shouldCreateAgreement() {
         assertThat(agreementResponseFromService.getAgreementId(), is(AGREEMENT_ID));
     }
 
+    @Test
+    void shouldThrowExceptionIfConnectorReturnsRecurringPaymentsNotAllowedExceptionWhenCreatingAgreement() {
+        when(mockConnectorUriGenerator.getAgreementURI(mockAccount)).thenReturn(AGREEMENTS_CONNECTOR_URI);
+        when(mockClient.target(AGREEMENTS_CONNECTOR_URI)).thenReturn(mockWebTarget);
+        when(mockWebTarget.request()).thenReturn(mockInvocationBuilder);
+        when(mockInvocationBuilder.accept(MediaType.APPLICATION_JSON)).thenReturn(mockInvocationBuilder);
+
+        CreateAgreementRequestParams createAgreementRequestParams = aCreateAgreementRequestParams()
+                .withReference(REFERENCE_ID)
+                .build();
+        when(mockInvocationBuilder.post(json(agreementPayload(createAgreementRequestParams)))).thenReturn(mockConnectorResponse);
+
+        when(mockConnectorResponse.getStatus()).thenReturn(HttpStatus.SC_UNPROCESSABLE_ENTITY);
+        CreateAgreementRequest agreementCreateRequest = new CreateAgreementRequest(CreateAgreementRequestBuilder
+                .builder().reference(REFERENCE_ID));
+        assertThrows(CreateAgreementException.class, () -> underTest.create(mockAccount, agreementCreateRequest));
```

Comment:
It looks like these tests will be pact tests eventually. Looking at the similar tests in `CreatePaymentServiceTest` we assert that the error identifier is passed through correctly from the connector response, and is the error identifier we expect in this scenario.

However, as this isn't currently a Pact test if we added that assertion we probably wouldn't be asserting much. So could possibly leave adding that assertion until we convert to pact tests..

Reply:
ok thanks for that

Reply:
no changes made here, leave for pact tests as suggested